File: app/schemas/cuenta.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel


class CuentaActualRespuesta(BaseModel):
    cuenta_id: int
    plan: Literal["profesional", "consultorio", "centro"]
    subscription_status: Literal["trial", "active", "past_due", "cancelled", "expired"]
    trial_started_at: datetime | None
    trial_ends_at: datetime | None
    trial_days_remaining: int

File: frontend/src/types/cuenta.ts
export type EstadoSuscripcion = "trial" | "active" | "past_due" | "cancelled" | "expired";
export type PlanCode = "profesional" | "consultorio" | "centro";
export interface CuentaActual { cuenta_id: number; plan: PlanCode; subscription_status: EstadoSuscripcion; trial_started_at: string | null; trial_ends_at: string | null; trial_days_remaining: number; }
